shared/cluster.go

```diff
@@ -200,6 +204,23 @@ func FetchClusterIP(namespace string, serviceName string) (string, string, error
 	return ip, port, err
 }
 
+// FetchClusterIPs returns the cluster IPs and port of the service.
+func FetchClusterIPs(namespace string, svc string) (string, string, error) {
```

Comment:
we already had this func, why adding another ?

Reply:
Its not the same, we have a func called `FetchClusterIP`... this func is calling `FetchClusterIPs` and if you see the implementation, they make look same but they are calling different resources

Reply:
ok then, so parametrize it to have only one.

Reply:
as per private discussion MD will implement a better approach that he provided.
create a generic function that acts based on a key to get something from spec.